### PR TITLE
Slack Notifier For Coverage

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/38540

## Description
Adding message to slack nightly of coverage percent of content

### CODE REVIEW

File: Tests/scripts/slack_notifier.py
Comment (dantavori) [inline Tests/scripts/slack_notifier.py:169]:
```suggestion
        unittests_fields: Optional[List[Dict]] = get_entities_fields(f'Failed Unittests - {len(artifact_data)}', artifact_data)
```

File: Tests/scripts/slack_notifier.py
Comment (dantavori) [inline Tests/scripts/slack_notifier.py:200]: let's set different colors for different ranges, e.g. 0%-49% danger, 50%-59% warning, 60%-100% good.

File: Tests/scripts/slack_notifier.py
Comment (dantavori) [inline Tests/scripts/slack_notifier.py:216]: only if `failed_entities` is not None

File: Tests/scripts/slack_notifier.py
Comment (dantavori) [inline Tests/scripts/slack_notifier.py:269]: same here

**Review by dantavori — CHANGES_REQUESTED** [review]
nice work 👍🏽

File: Tests/scripts/gitlab_slack_notifier.py
Comment (dantavori) [inline Tests/scripts/gitlab_slack_notifier.py:52]: why do we have this in gitlab build if we don't run unit tests there?

Comment (avidan-H) [reply Tests/scripts/gitlab_slack_notifier.py:52]: We do run unit tests there in various flows.

File: Tests/scripts/slack_notifier.py
Comment (avidan-H) [inline Tests/scripts/slack_notifier.py:125]: Probably be more accurate to describe it as the relative file path to the artifacts directory.

File: Tests/scripts/slack_notifier.py
Comment (avidan-H) [inline Tests/scripts/slack_notifier.py:128]:
```suggestion
        file_name = os.path.join(ARTIFACTS_FOLDER, artifact_suffix)
```

**Review by avidan-H — CHANGES_REQUESTED** [review]

File: Tests/scripts/gitlab_slack_notifier.py
Comment (tomneeman151293) [reply Tests/scripts/gitlab_slack_notifier.py:55]: Changed to failing_unit_tests = failing_unit_tests.split('\n')
 
my bad.

File: Tests/scripts/slack_notifier.py
Comment (avidan-H) [inline Tests/scripts/slack_notifier.py:215]: Change `dict()` to `{}` and you should provide a default value to the `.get('@line-rate')` otherwise you could have the situation where you are trying to multiply a `None type` by `100.0`. If you want there to be an error of some kind if no `@line-rate` field is present then explicitly do so with a helpful message.

File: Tests/scripts/slack_notifier.py
Comment (avidan-H) [inline Tests/scripts/slack_notifier.py:221]: Do we need this line if the the value is an empty list?

**Review by avidan-H — CHANGES_REQUESTED** [review]

File: Tests/scripts/slack_notifier.py
Comment (tomneeman151293) [reply Tests/scripts/slack_notifier.py:136]: This was the behaviour before so i didnt want to change it. This is needed because the artifact will not exist if no test have failed if I remember correctly

File: Tests/scripts/slack_notifier.py
Comment (tomneeman151293) [reply Tests/scripts/slack_notifier.py:215]: Added logging messages if coverage file or line rate data are not found

File: Tests/scripts/slack_notifier.py
Comment (tomneeman151293) [reply Tests/scripts/slack_notifier.py:221]: Not sure to be honest. But it does work this way as expected. I can perform checks without it if you think it is better without.

Comment (avidan-H) [reply Tests/scripts/slack_notifier.py:221]: Not a big deal. Can leave it as it is.

**Review by avidan-H — APPROVED** [review]

**Comment by tomneeman151293** [comment]
@dantavori Anything left from your side here? Cant merge until you approve

**Review by dantavori — APPROVED** [review]

**Pull request merged**
